.evergreen/config.in.yml

```diff
@@ -130,123 +130,40 @@ functions:
           DRIVERS_TOOLS: ${DRIVERS_TOOLS}
 
   "run tests":
-    - command: shell.exec
-      type: test
+    - command: ec2.assume_role
       params:
-        silent: true
-        working_dir: "src"
-        script: |
-          if [ -n "${CLIENT_ENCRYPTION}" ]; then
-            cat <<EOT > prepare_client_encryption.sh
-            export CLIENT_ENCRYPTION=${CLIENT_ENCRYPTION}
-            export RUN_WITH_MONGOCRYPTD=${RUN_WITH_MONGOCRYPTD}
-            export CSFLE_KMS_PROVIDERS='${CSFLE_KMS_PROVIDERS}'
-            export FLE_AWS_KEY2='${FLE_AWS_KEY2}'
-            export FLE_AWS_SECRET2='${FLE_AWS_SECRET2}'
-            export AWS_ACCESS_KEY_ID='${AWS_ACCESS_KEY_ID}'
-            export AWS_SECRET_ACCESS_KEY='${AWS_SECRET_ACCESS_KEY}'
-            export AWS_REGION='${AWS_REGION}'
-            export AWS_CMK_ID='${AWS_CMK_ID}'
-            export AWS_DEFAULT_REGION='us-east-1'
-            export KMIP_TLS_CA_FILE="${DRIVERS_TOOLS}/.evergreen/x509gen/ca.pem"
-            export KMIP_TLS_CERT_FILE="${DRIVERS_TOOLS}/.evergreen/x509gen/client.pem"
-          EOT
-          fi
-    - command: shell.exec
+        role_arn: ${OIDC_AWS_ROLE_ARN}
```

Comment:
This is actually the general secrets manager role.  If we want, I can change the name of the secret in Evergreen and adjust the config, but I chose not to because this PR is already quite large.